### PR TITLE
add rb-tty-prompt

### PR DESCRIPTION
**Pull request by rluria14**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by rluria14** [comment]
@sodre @adbeda agree to be maintainers?

File: recipes/rb-tty-prompt/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+source:
+  url: https://rubygems.org/downloads/{{ name }}-{{ version }}.gem
+  sha256: 3347d3535aaabd71952e52b0936f523ce77d0b345b1649b733dd9a9a92d8e157
+
+build:
+  noarch: generic
+  number: 0
+  script:
+    - gem install --norc -l -V --ignore-dependencies {{ name }}-{{ version }}.gem
+    - gem unpack {{ name }}-{{ version }}.gem
```
Comment (sodre) [inline recipes/rb-tty-prompt/meta.yaml]:
```suggestion
      && gem unpack {{ name }}-{{ version }}.gem
```

**Comment by rluria14** [comment]
pending on rb-pastel getting added into conda-forge.

File: recipes/rb-tty-prompt/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    - gem install --norc -l -V --ignore-dependencies {{ name }}-{{ version }}.gem
+      && gem unpack {{ name }}-{{ version }}.gem
+  skip: {{ win }}
+
+requirements:
+  host:
+    - ruby
+  run:
+    - ruby
+    - rb-necromancer >=0.5,<1
```
Comment (sodre) [inline recipes/rb-tty-prompt/meta.yaml]: The constraint on rubygems was `~>0.5.0`,
```suggestion
    - rb-necromancer >=0.5.0,<0.6
```

File: recipes/rb-tty-prompt/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    - rb-pastel >=0.7.3,<1
```
Comment (sodre) [inline recipes/rb-tty-prompt/meta.yaml]: the constraint was `~>0.7.0`
```suggestion
    - rb-pastel >=0.7.0,<0.8
```

File: recipes/rb-tty-prompt/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    - rb-tty-reader >=0.6,<1
```
Comment (sodre) [inline recipes/rb-tty-prompt/meta.yaml]: the constraint  was `~>0.6.0`
```suggestion
    - rb-tty-reader >=0.6.0,<0.7
```

File: recipes/rb-tty-prompt/meta.yaml (hunk 3 above)
Comment (rluria14) [reply recipes/rb-tty-prompt/meta.yaml]: Thanks very much. I have committed your requested changes.

Comment (rluria14) [reply recipes/rb-tty-prompt/meta.yaml]: i still seem to be getting a version error with `rb-pastel >=0.7.0,<0.8`

**Comment by sodre** [comment]
@rluria14, the rb-pastel package never built, probably because of some racing condition with the build-servers. 

I still need you to correct the rb-pastel constraint. It should match what is in rubygems.

**Comment by rluria14** [comment]
> @rluria14, the rb-pastel package never built, probably because of some racing condition with the build-servers.
> 
> I still need you to correct the rb-pastel constraint. It should match what is in rubygems.

rb-pastel constraint has been updated and all checks have now passed.

**Comment by sodre** [comment]
the ruby requirements are not needed at the moment, but thanks for adding it.

**Pull request merged**